// react-numpad/build/elements/Display.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _backspace = _interopRequireDefault(require("react-icons/lib/md/backspace"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n  flex-grow: 1;\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n  &:read-only {\n    cursor: not-allowed;\n  }\n  border-radius: 0px;\n  cursor: default;\n  background: transparent;\n  font-size: 1.3em;\n  outline: none;\n  border: none;\n  width: 100%;\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n  background: none;\n  cursor: default;\n  border: none;\n  outline: none;\n  font-size: 1.6em;\n  padding: 0px 2px 0px 0px;\n  color: ", ";\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  display: flex;\n  padding: 2px 5px 5px 8px;\n  align-items: center;\n  border: none;\n  background: white;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var Wrapper = _styledComponents.default.div(_templateObject());

var Backspace = _styledComponents.default.button(_templateObject2(), function (props) {
  return props.theme.header.primaryColor;
});

var Input = _styledComponents.default.input(_templateObject3());

var Display = _styledComponents.default.div(_templateObject4());

var DisplayWrapper = function DisplayWrapper(_ref) {
  var value = _ref.value,
      displayRule = _ref.displayRule,
      dateFormat = _ref.dateFormat,
      cancel = _ref.cancel;
  return _react.default.createElement(Wrapper, null, _react.default.createElement(Display, null, _react.default.createElement(Input, {
    value: displayRule(value, dateFormat),
    readOnly: true,
    autoFocus: true
  })), _react.default.createElement(Backspace, {
    onClick: cancel
  }, _react.default.createElement(_backspace.default, null)));
};

DisplayWrapper.propTypes = {
  value: _propTypes.default.string.isRequired,
  displayRule: _propTypes.default.func.isRequired,
  dateFormat: _propTypes.default.string,
  cancel: _propTypes.default.func
};
DisplayWrapper.defaultProps = {
  dateFormat: 'MM/DD/YYYY',
  cancel: function cancel() {}
};
var _default = DisplayWrapper;
exports.default = _default;
