// react-numpad/build/components/CalendarDate.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _dateRange = _interopRequireDefault(require("react-icons/lib/md/date-range"));

var _moment = _interopRequireDefault(require("moment"));

var _elements = require("../elements");

var _date = _interopRequireDefault(require("../utils/date"));

var _NumPad = _interopRequireDefault(require("./NumPad"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var validation = function validation() {
  return true;
};

var formatInputValue = function formatInputValue(value) {
  return value.toString().replace(/\D+/g, '');
};

var keyValid = function keyValid(value, min, max, dateFormat) {
  var validAfter = min ? value.isSameOrAfter((0, _moment.default)(min, dateFormat)) : true;
  var validBefore = max ? value.isSameOrBefore((0, _moment.default)(max, dateFormat)) : true;
  return validAfter && validBefore;
};

var displayRule = function displayRule() {
  var value = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : '';
  var dateFormat = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 'MM/DD/YYYY';
  return _date.default.display(value, dateFormat);
};

var inputButtonContent = _react.default.createElement(_dateRange.default, null);

var _default = (0, _NumPad.default)({
  element: _elements.Calendar,
  keyValid: keyValid,
  formatInputValue: formatInputValue,
  validation: validation,
  displayRule: displayRule,
  inputButtonContent: inputButtonContent
});

exports.default = _default;
