// react-numpad/build/components/CalendarAppointment.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _dateRange = _interopRequireDefault(require("react-icons/lib/md/date-range"));

var _elements = require("../elements");

var _NumPad = _interopRequireDefault(require("./NumPad"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var validation = function validation() {
  return true;
};

var keyValid = function keyValid() {
  return true;
};

var formatInputValue = function formatInputValue(value) {
  return value;
};

var displayRule = function displayRule(value) {
  return value;
};

var inputButtonContent = _react.default.createElement(_dateRange.default, null);

var _default = (0, _NumPad.default)({
  element: _elements.Appointment,
  keyValid: keyValid,
  formatInputValue: formatInputValue,
  validation: validation,
  displayRule: displayRule,
  inputButtonContent: inputButtonContent
});

exports.default = _default;
